src/ajax/check_hgvs_dialogue.php

```diff
@@ -0,0 +1,603 @@
+        //  should add the information into the genomic mappings.
+        $aMappedVariant['data']['genomic_mappings'][$sGenomeBuildID] = $aMappedVariant['data']['DNA'];
+    }
+
+    // Returning the mapping for transcript, RNA and protein variants.
+    foreach ($aMappedVariant['data']['transcript_mappings'] as $sTranscript => $aTranscriptData) {
+        // Filling in the input fields.
+        print('
+        // Adding transcript info to the fields.        
+        var oTranscriptField = $("input").filter(function() { 
+            return $(this).data("id_ncbi") == "' . $sTranscript . '" 
+        });
+        
+        if (!oTranscriptField.prop("disabled")) {
+            oTranscriptField.val("' . $aTranscriptData['DNA'] . '").attr("class", "acc");;
+            oTranscriptField.siblings("img:first").attr({src: "gfx/check.png", title: "Validated"}).show();
+            var sBaseOfFieldName = oTranscriptField.attr("name").substring(0, oTranscriptField.attr("name").indexOf("DNA"));
+            $(\'#variantForm input[name$="\' + sBaseOfFieldName + "RNA" + \'"]\').val("' . $aTranscriptData['RNA'] . '").attr("class", "acc");
+            $(\'#variantForm input[name$="\' + sBaseOfFieldName + "Protein" + \'"]\').val("' . $aTranscriptData['protein'] . '").attr("class", "acc");
+        }');
+    }
+
+    // Returning the mapping for genomic variants.
+    foreach ($aMappedVariant['data']['genomic_mappings'] as $sGBID => $aMappedGenomicVariant) {
+        if (is_string($aMappedGenomicVariant)) {
+            // The variant is a string. We don't need to make any edits.
+            $sMappedGenomicVariant = $aMappedGenomicVariant;
+
+        } elseif (count($aMappedGenomicVariant) == 1) {
+            // The variant is formatted as an array, since multiple variants were
+            //  possible. However, only one variant was found. We can simply take
+            //  the first element.
+            $sMappedGenomicVariant = $aMappedGenomicVariant[0];
+
+        } else {
+            // Multiple possible variants were found. We will inform the user,
+            //  and concatenate the variants similarly to the example below:
+            // NC_1233456.1:g.1del + NC_123456.1:g.2_3del + NC_123456.1:g.4del =
+            //  NC_123456.1:g.1del^2_3del^4del.
+            update_dialogue('<br>There were multiple genomic variant predictions for build ' . $sGBID . '.');
+
+            $sMappedGenomicVariant =
+                preg_replace('/(.*:[a-z]\.).*/', '', $aMappedGenomicVariant[0]) .
+                implode('^',
+                    array_map(function ($sFullVariant) {
+                        return preg_replace('/.*:[a-z]\./', '', $sFullVariant);
+                    }, $aMappedGenomicVariant)
+                );
+        }
+
+        // Removing the reference sequence.
+        $sMappedGenomicVariant = preg_replace('/.*:/', '', $sMappedGenomicVariant); // Fixme; Find a cleaner way of cutting off the reference sequence.
```

Comment:
Personally, I always use `strstr()` with a true in argument 3. I guess the same could be used in the `preg_replace()` above. Since we know we have `refseq:prefix.variant` there, we could simply do a `strstr()` on a period (".") to merge the multiple variant descriptions.